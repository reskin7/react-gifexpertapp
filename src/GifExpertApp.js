import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Overwatch']);
  const [quantity, setQuantity] = useState(1);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} setQuantity={setQuantity} />
      <hr />

      <ol>
        {categories.map((cats) => (
          <GifGrid key={cats} category={cats} quantity={quantity} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
