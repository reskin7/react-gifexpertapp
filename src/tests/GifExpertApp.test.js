import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';

describe('Test on GifExpertApp', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show an array of categories', () => {
    const categories = ['One punch', 'Dragon ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
