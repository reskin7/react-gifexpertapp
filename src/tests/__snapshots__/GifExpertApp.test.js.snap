// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifExpertApp should match snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
    setQuantity={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test on GifExpertApp should show an array of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
    setQuantity={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
      quantity={1}
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
      quantity={1}
    />
  </ol>
</Fragment>
`;
